tests/test_testing.py

```diff
@@ -26,3 +26,10 @@ def test_assert_yaml_fails(caplog):
     obs = log[2].replace("\n", "")  # strip newlines from observed log message
     exp = r"Error parsing file while scanning a simple key.*, line 4, column 1"
     assert re.match(exp, obs)
+
+
+def test_special_character():
+    """Check that a non-printable character in any yaml file will raise an error"""
+    match = "variables.yaml, line 5, col 53."
+    with pytest.raises(ValueError, match=match):
+        assert_valid_yaml(TEST_DATA_DIR / "special_character")
```

Comment:
I would also rename the folder where the test data are located
```suggestion
        assert_valid_yaml(TEST_DATA_DIR / "hidden_character")
```